File: energiai-frontend/src/app/core/models/historial-response.ts
import { RecomendacionDetalle } from './analisis-response';

export interface HistorialResponse {
  id: number;
  creadoEn: string;
  usuario: string;
  consumoKwh: number;
  tipoInmueble: string;
  cantidadEquipos: number;
  horasAltoConsumo: number;
  usoHorarioPico: boolean;
  categoria: string;
  probabilidad: number;
  costo_estimado_mensual: number;
  recomendaciones: string[];
  clasificacion_equipos?: Record<string, number>;
  recomendaciones_detalle?: RecomendacionDetalle[];
  origen_prediccion?: string;
  modelo_version?: string;
  advertencias?: string[];
  nombre_o_numero_analisis?: string;
}

File: energiai-api/src/main/java/com/hackathon/energiai_api/DTOs/HistorialResponse.java
package com.hackathon.energiai_api.DTOs;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

public record HistorialResponse(
        Long id,
        LocalDateTime creadoEn,
        String usuario,
        Integer consumoKwh,
        String tipoInmueble,
        Integer cantidadEquipos,
        Integer horasAltoConsumo,
        Boolean usoHorarioPico,
        String categoria,
        BigDecimal probabilidad,
        BigDecimal costo_estimado_mensual,
        List<String> recomendaciones,
        Map<String, Integer> clasificacion_equipos,
        List<RecomendacionModelo> recomendaciones_detalle,
        String origen_prediccion,
        String modelo_version,
        List<String> advertencias,
        String nombre_o_numero_analisis
) {
}
